src/Type/Doctrine/QueryBuilder/QueryBuilderGetQueryDynamicReturnTypeExtension.php

```diff
@@ -152,6 +153,18 @@ public function getTypeFromMethodCall(
 					continue;
 				}
 
+				if ($lowerMethodName === 'set') {
+					try {
+						$args = $this->argumentsProcessor->processArgs($scope, $methodName, array_slice($calledMethodCall->getArgs(), 0, 1));
+					} catch (DynamicQueryBuilderArgumentException $e) {
+						return $defaultReturnType;
+					}
+					if (count($args) === 1) {
+						$queryBuilder->set($args[0], $args[0]);
```

Comment:
This branch does not seem to be tested

Reply:
It is. This is the main thing that enables the whole thing. Maybe you missed array_slice above.

Reply:
Oh alright, I know less and less about this extension every day :D